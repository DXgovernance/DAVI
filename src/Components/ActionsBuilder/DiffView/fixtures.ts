export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
