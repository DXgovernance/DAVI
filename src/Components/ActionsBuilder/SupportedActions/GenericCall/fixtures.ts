import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'Components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const genericDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.GENERIC_CALL,
  function: ERC20Contract.getFunction('increaseAllowance'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    param1: BigNumber.from(10000),
    param2: BigNumber.from(10000),
    param3: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
