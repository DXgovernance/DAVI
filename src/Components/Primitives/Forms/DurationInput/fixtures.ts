export const defaultProps = {
  value: 0,
  onChange: () => jest.fn(),
};
