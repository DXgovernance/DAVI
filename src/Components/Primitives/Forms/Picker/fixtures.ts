import { PickerProps } from './types';

export const fullProps: PickerProps = {
  data: [
    {
      id: '0',
      title: 'Different',
      subtitle: 'There are no references here',
      value: '0',
    },
    {
      id: '1',
      title: 'Token1',
      subtitle: 'This is the token one',
      value: 'value token 1',
      balance: 135,
    },
    {
      id: '2',
      title: 'Token2',
      value: 'value token 2',
    },
    {
      id: '3',
      title: 'Token3',
      subtitle: 'Some subtitle',
      value: 'value token 3',
    },
    {
      id: '4',
      title: 'Token4',
      value: 'value token 4',
    },
    {
      id: '5',
      title: 'Token5',
      value: 'value token 5',
      balance: 350,
    },
    {
      id: '6',
      title: 'Token6',
      subtitle: 'Indeed this is sub-text',
      value: 'value token 6',
    },
  ],
  header: 'Pick a token',
  isOpen: true,
  onSelect: option => console.log(option),
  onClose: () => {},
};

export const swaprPairs: PickerProps = {
  isOpen: true,
  onSelect: option => console.log(option),
  onClose: () => {},
  header: 'Pick a pair',
  data: [
    {
      title: 'TORN-WETH',
      address: '0x010946a9189f4fa8d6bf01d2fa32e0edf3d6e03b',
      reserve0: '21.979143344333184701',
      reserve1: '0.295076965839191706',
      reserveUSD: '569.2364021717024122251006928140325',
      reserveNativeCurrency: '0.295076965839191706',
      totalSupply: '2.499685243646281452',
      token0: {
        address: '0x77777feddddffc19ff86db637967013e6c6a116c',
        name: 'TornadoCash',
        symbol: 'TORN',
        decimals: '18',
      },
      token1: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'STAKE-DXD',
      address: '0x04e9644c7b5c82280fe95f515b7c4ad18a72db83',
      reserve0: '0.321291693076585919',
      reserve1: '0.020502751836012276',
      reserveUSD: '15.32220501881621236926362444314866',
      reserveNativeCurrency: '0.007406919217920272769253686646761754',
      totalSupply: '0.080549044622494839',
      token0: {
        address: '0x0ae055097c6d159879521c384f1d2123d1f195e6',
        name: 'STAKE',
        symbol: 'STAKE',
        decimals: '18',
      },
      token1: {
        address: '0xa1d65e8fb6e87b60feccbc582f7f97804b725521',
        name: 'DXdao',
        symbol: 'DXD',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'DAI-USDC',
      address: '0x0e18852ee5ce266e2ce6f4844ba04cd9cd11af5b',
      reserve0: '0.000083384608530149',
      reserve1: '0.000084',
      reserveUSD: '0.0001632092461365910476688685934587097',
      reserveNativeCurrency: '0.000000264969183078724715385506512724458',
      totalSupply: '0.000000000083420641',
      token0: {
        address: '0x6b175474e89094c44da98b954eedeac495271d0f',
        name: 'Dai Stablecoin',
        symbol: 'DAI',
        decimals: '18',
      },
      token1: {
        address: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
        name: 'USD Coin',
        symbol: 'USDC',
        decimals: '6',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'TCAP-CTX',
      address: '0x15d37a2c7e4a88adb0ef5a6c85866a54d261201d',
      reserve0: '0.004363640935513935',
      reserve1: '0.087746620703447774',
      reserveUSD: '0',
      reserveNativeCurrency: '0',
      totalSupply: '0.019496932818319274',
      token0: {
        address: '0x16c52ceece2ed57dad87319d91b5e3637d50afa4',
        name: 'TCAP Token',
        symbol: 'TCAP',
        decimals: '18',
      },
      token1: {
        address: '0x321c2fe4446c7c963dc41dd58879af648838f98d',
        name: 'Cryptex',
        symbol: 'CTX',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'USDC-CRV',
      address: '0x1d9e9118c8ac62360364b659035bd87d110691e7',
      reserve0: '0.095886',
      reserve1: '0.10154694589503153',
      reserveUSD: '0.2260850332568574529014009796904103',
      reserveNativeCurrency: '0.0002011286020137123791924221230501354',
      totalSupply: '0.000000098565255478',
      token0: {
        address: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
        name: 'USD Coin',
        symbol: 'USDC',
        decimals: '6',
      },
      token1: {
        address: '0xd533a949740bb3306d119cc777fa900ba034cd52',
        name: 'Curve DAO Token',
        symbol: 'CRV',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'YMTPP-WETH',
      address: '0x26661b84ec077cd4ce9876657fb35c4d9b0bb0f9',
      reserve0: '673418.539295465611922907',
      reserve1: '0.038159528779165558',
      reserveUSD: '148.8992051270776033411438634668967',
      reserveNativeCurrency: '0.038159528779165558',
      totalSupply: '160.277670087204858882',
      token0: {
        address: '0x61dad5b3c38bb5963b13b325be52ed28ffdf6343',
        name: 'YAMATOPEPE',
        symbol: 'YMTPP',
        decimals: '18',
      },
      token1: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'NOAHP-WETH',
      address: '0x3ad1754fd7b22af9f9ce66a068e146749aae2d61',
      reserve0: '46235976.082820852285269257',
      reserve1: '0.033801712512721425',
      reserveUSD: '99.66386402218806197538671174414678',
      reserveNativeCurrency: '0.033801712512721425',
      totalSupply: '1241.099995971315760731',
      token0: {
        address: '0x41b3f18c6384dc9a39c33afeca60d9b8e61eaa9f',
        name: 'NOAH DECENTRALIZED STATE COIN',
        symbol: 'NOAHP',
        decimals: '18',
      },
      token1: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'RAD-WETH',
      address: '0x3fff4a26a98b730c9027d7ab70a2e89916cacd52',
      reserve0: '0.172037299461729311',
      reserve1: '0.000545603308030155',
      reserveUSD: '3.400794002132388789368633737174888',
      reserveNativeCurrency: '0.00109120661606031',
      totalSupply: '0.009599385258801677',
      token0: {
        address: '0x31c8eacbffdd875c74b94b077895bd78cf1e64a3',
        name: 'Radicle',
        symbol: 'RAD',
        decimals: '18',
      },
      token1: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'WETH-PAN',
      address: '0x46575017362add18630dd99246bca8f853b80ede',
      reserve0: '0.001207920729132939',
      reserve1: '47.448993205213036559',
      reserveUSD: '9.379555947938372930418567245075574',
      reserveNativeCurrency: '0.002415841458265878',
      totalSupply: '0.23709933895873348',
      token0: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      token1: {
        address: '0xd56dac73a4d6766464b38ec6d91eb45ce7457c44',
        name: 'Panvala pan',
        symbol: 'PAN',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
  ],
};
