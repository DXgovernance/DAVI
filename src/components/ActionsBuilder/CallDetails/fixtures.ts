import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
