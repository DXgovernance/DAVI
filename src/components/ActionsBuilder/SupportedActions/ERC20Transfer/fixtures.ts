import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const erc20TransferEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const erc20TransferDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
  value: BigNumber.from(0),
  args: {
    _to: '0xaF8eB8C3A5d9d900AA0B98e3Df0bcC17d3C5F698',
    _value: BigNumber.from('5000000000000000000'),
  },
};
