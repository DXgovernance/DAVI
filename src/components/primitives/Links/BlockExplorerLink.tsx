import { Segment } from 'components/ActionsBuilder/SupportedActions/common/infoLine';
import { Avatar } from 'components/Avatar';
import useENSAvatar from 'hooks/Guilds/ens/useENSAvatar';
import { useERC20Info } from 'hooks/Guilds/erc20/useERC20Info';
import { useTokenList } from 'hooks/Guilds/tokens/useTokenList';
import { useMemo } from 'react';
import { MAINNET_ID, shortenAddress } from 'utils';
import { resolveUri } from 'utils/url';
import { Chain, useNetwork } from 'wagmi';
import { Flex } from '../Layout';
import { ExternalLink } from './ExternalLink';
import { BlockExplorerLinkProps } from './types';

const getBlockExplorerUrl = (
  chain: Chain,
  address: string,
  type: 'address' | 'tx'
) => {
  if (!chain || !chain?.blockExplorers?.default) return null;

  return `${chain.blockExplorers.default.url}/${type}/${address}`;
};

export const BlockExplorerLink: React.FC<BlockExplorerLinkProps> = ({
  address,
  showAvatar,
  shortAddress = false,
  avatarSize = 24,
  disableLink = false,
}) => {
  const { chain } = useNetwork();
  const { ensName, imageUrl } = useENSAvatar(address, MAINNET_ID);
  const { data: erc20Info } = useERC20Info(address);
  const { tokens } = useTokenList(chain.id);

  const detailedTokenData = useMemo(() => {
    if (!erc20Info || !tokens) return null;

    return tokens.find(
      ({ address: detailedTokenAddress }) => detailedTokenAddress === address
    );
  }, [tokens, erc20Info, address]);

  if (!address) return null;

  const blockExplorerUrl = getBlockExplorerUrl(chain, address, 'address');

  return (
    <Flex
      direction="row"
      justifyContent="left"
      data-testid="block-explorer-container"
    >
      {showAvatar && (
        <Segment>
          <Avatar
            defaultSeed={address}
            src={resolveUri(detailedTokenData?.logoURI) || imageUrl}
            size={avatarSize}
            data-testid="avatar"
          />
        </Segment>
      )}
      {disableLink ? (
        <div>
          {erc20Info?.symbol ||
            ensName ||
            (shortAddress ? shortenAddress(address) : address)}
        </div>
      ) : (
        <ExternalLink href={blockExplorerUrl} data-testid="external-link">
          {erc20Info?.symbol ||
            ensName ||
            (shortAddress ? shortenAddress(address) : address)}
        </ExternalLink>
      )}
    </Flex>
  );
};
