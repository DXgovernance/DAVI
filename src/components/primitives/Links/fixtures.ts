export const MOCK_ADDRESS = '0x0000000000000000000000000000000000000000';
export const MOCK_ENS_NAME = 'name.eth';
