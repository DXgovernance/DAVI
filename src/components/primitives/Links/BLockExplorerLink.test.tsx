import { mockBigNumber } from 'components/ActionsBuilder/SupportedActions/SetGuildConfig/fixtures';
import { render } from 'utils/tests';
import { BlockExplorerLink } from './BlockExplorerLink';

jest.mock('wagmi', () => ({
  useNetwork: () => ({
    chain: {
      id: 1,
      blockExplorers: {
        default: {
          url: 'https://etherscan.io/',
        },
      }.default.url,
    },
  }),
  useEnsAddress: () => ({
    data: '0x0000000000000000000000000000000000000000',
  }),
  useEnsName: () => ({
    data: 'name.eth',
  }),
  useEnsResolver: () => ({
    data: {
      name: 'naaame.eth',
      address: '0x0000000000000000000000000000000000000000',
    },
  }),
  useContractRead: () => ({
    data: 'e30101701220e09973e8c9e391cb063bd6654356e64e0ceced7858a29a8c01b165e30a5eb5be',
  }),
  useContractReads: () => ({
    data: [{}],
  }),
}));

jest.mock('hooks/Guilds/tokens/useTokenList', () => ({
  useTokenList: () => ({
    tokens: [],
  }),
}));

jest.mock('hooks/Guilds/erc20/useERC20Info', () => ({
  useERC20Info: () => ({
    name: 'Test ERC20',
    symbol: 'TEST',
    decimals: 18,
    totalSupply: mockBigNumber,
  }),
}));

describe('BlockExplorerLink', () => {
  describe('BlockExplorerLink render', () => {
    it('Should match snapshot', () => {
      const { container } = render(
        <BlockExplorerLink
          address={'0x4e91c9F086DB2Fd8aDb1888e9b18e17F70B7BdB6'}
        />
      );
      expect(container).toMatchSnapshot();
    });
    it('should not render if address is an empty string, null or undefined', () => {
      let queryByTestIdEmptyStringAddress;
      let queryByTestIdUndefinedAddress;
      let queryByTestIdNullAddress;

      let componentWithEmptryStringAddress = render(
        <BlockExplorerLink address={''} />
      );
      queryByTestIdEmptyStringAddress =
        componentWithEmptryStringAddress.queryByTestId;
      let componentWithNullAddress = render(
        <BlockExplorerLink address={null} />
      );
      queryByTestIdNullAddress = componentWithNullAddress.queryByTestId;
      let componentWithUndefinedAddress = render(
        <BlockExplorerLink address={undefined} />
      );
      queryByTestIdUndefinedAddress =
        componentWithUndefinedAddress.queryByTestId;

      expect(
        queryByTestIdEmptyStringAddress('block-explorer-container')
      ).toBeNull();
      expect(queryByTestIdNullAddress('block-explorer-container')).toBeNull();
      expect(
        queryByTestIdUndefinedAddress('block-explorer-container')
      ).toBeNull();
    });
  });

  describe('BlockExplorerLink with Avatar', () => {
    it('should show avatar when receives showAvatar prop', () => {
      let getByTestId;
      let componentWithAvatar = render(
        <BlockExplorerLink
          address={'0x4e91c9F086DB2Fd8aDb1888e9b18e17F70B7BdB6'}
          showAvatar
        />
      );
      getByTestId = componentWithAvatar.getByTestId;
      const container = getByTestId('avatar');
      expect(container).toBeDefined();
    });
    it('should not show avatar when does not receive showAvatar prop', () => {
      let queryByTestId;
      let componentWithAvatar = render(
        <BlockExplorerLink
          address={'0x4e91c9F086DB2Fd8aDb1888e9b18e17F70B7BdB6'}
        />
      );
      queryByTestId = componentWithAvatar.queryByTestId;
      expect(queryByTestId('avatar')).toBeNull();
    });
  });
});
