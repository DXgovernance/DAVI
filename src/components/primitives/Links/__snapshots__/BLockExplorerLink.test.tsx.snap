// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockExplorerLink BlockExplorerLink render Should match snapshot 1`] = `
.c2 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c0 {
  display: Flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  border-radius: 10px;
  margin: 0;
  padding: 0;
}

.c1 {
  color: #A1A6B0;
  margin-right: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-wrap: break-word;
}

.c1:hover {
  cursor: pointer;
  color: #fff;
}

.c3 {
  margin-right: 0.5rem;
  overflow-wrap: break-word;
}

<div>
   
  <div
    class="c0"
    data-testid="block-explorer-container"
    direction="row"
  >
    <a
      class="c1"
      rel="noopener"
      target="_blank"
    >
      <div
        class="c2 c3"
      >
        name.eth
      </div>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
        />
        <polyline
          points="15 3 21 3 21 9"
        />
        <line
          x1="10"
          x2="21"
          y1="14"
          y2="3"
        />
      </svg>
    </a>
  </div>
   
</div>
`;
